Clients/GUI/Pages/InstrumentInfo.razor

```diff
@@ -0,0 +1,469 @@
+            {
+                UpdateBalanceVariable();
+            }
+        }
+        catch
+        {
+            userBalance = oldBalance;
+            errorOccured = true;
+            errorMessage = "Can't update balance, try again later.";
+        }
+    }
+
+    private async Task GetCountFromPortfolio()
+    {
+        var instrumentInfo = await BrokerManager.GetInstrumentFromPortfolio(userBalance.UserId, instrument.Figi);
+        countInPortfolio = instrumentInfo.TotalCount;
+    }
+
+    private async Task GetUserBalance()
+    {
+        userBalance = await BrokerManager.GetUserBalance(userBalance.UserId);
+        UpdateBalanceVariable();
+    }
+
+    private async Task Sell()
+    {
+        UpdateBalanceVariable();
+        if (countInPortfolio < lotsToTrade * instrument.CountInLot)
+            Modal.Show<EmptyModal>("Count in portfolio less than you can sell");
+        else
+        {
+            try
+            {
+                var result = await SendTradeRequest(OperationType.Sell);
+                if (result == false)
+                {
+                    errorOccured = true;
+                    errorMessage = "Unable to sell instrument. Try again later.";
+                }
+                else
+                {
+                    await GetCountFromPortfolio();
+                    await GetUserBalance();
+                }
+            }
+            catch (Exception e)
+            {
+                errorOccured = true;
+                errorMessage = e.Message;
+            }
+        }
+    }
+
+    private async Task Buy()
+    {
+        UpdateBalanceVariable();
+        if (currentBalance < lotsToTrade * instrument.CountInLot * price)
+            Modal.Show<EmptyModal>("You need more money to make a deal");
+        else
+        {
+            try
+            {
+                var result = await SendTradeRequest(OperationType.Buy);
+                if (result == false)
+                {
+                    errorOccured = true;
+                    errorMessage = "Unable to buy instrument. Try again later.";
+                }
+                else
+                {
+                    await GetCountFromPortfolio();
+                    await GetUserBalance();
+                }
+            }
+            catch (Exception e)
+            {
+                errorOccured = true;
+                errorMessage = e.Message;
+            }
+        }
+    }
+
+    private async Task<bool> SendTradeRequest(OperationType operation)
+    {
+        var request = new TradeRequest()
+        {
+            UserId = userBalance.UserId,
+            Broker = BrokerType.TinkoffBroker,
+            Token = tinkoffToken,
+            Operation = operation,
+            Figi = instrument.Figi,
+            Count = lotsToTrade * instrument.CountInLot,
+            Price = price,
+            Currency = currency
+        };
+
+        return await BrokerManager.Trade(request);
```

Comment:
There can't be any exceptions or why we don't catch anything here?

Reply:
I have an exception handle in the sell and buy method

Reply:
> I have an exception handle in the sell and buy method

And what if you get some connection exception?

Reply:
fixed